src/relcast.erl

```diff
@@ -262,17 +279,28 @@ deliver(Message, FromActorID, State = #state{key_count=KeyCount, db=DB, active_c
     end.
 
 -spec take(pos_integer, #state{}) -> {not_found, #state{}} | {ok, reference(), binary(), #state{}}.
-take(ForActorID, State = #state{bitfieldsize=BitfieldSize, db=DB}) ->
+take(ForActorID, State = #state{bitfieldsize=BitfieldSize, db=DB, module=Module}) ->
     %% we need to find the first "unacked" message for this actor
     %% we should remember the last acked message for this actor ID and start there
     %% check if there's a pending ACK and use that to find the "last" key, if present
     case maps:get(ForActorID, State#state.pending_acks, undefined) of
         {Ref, CF, Key, _Multicast} ->
             case rocksdb:get(DB, CF, Key, []) of
-                {ok, <<1:1/integer, ForActorID:15/integer, Value/binary>>} ->
+                {ok, <<1:2/integer, ForActorID:14/integer, Value/binary>>} ->
                     {ok, Ref, Value, State};
-                {ok, <<0:1/integer, _:(BitfieldSize)/integer, Value/binary>>} ->
+                {ok, <<0:2/integer, _:(BitfieldSize)/integer, Value/binary>>} ->
                     {ok, Ref, Value, State};
+                {ok, <<2:2/integer, _:(BitfieldSize)/integer, Value/binary>>} ->
+                    case Module:callback_message(ForActorID, Value, State#state.modulestate) of
+                        none ->
+                            %% nothing for this actor, flip the bit
+                            ActorIDStr = ["-", integer_to_list(ForActorID+1)],
+                            ok = rocksdb:merge(State#state.db, CF, Key, list_to_binary(ActorIDStr), []),
+                            %% keep looking
+                            take(ForActorID, State#state{pending_acks=maps:remove(ForActorID, State#state.pending_acks)});
+                        Message ->
+                            {ok, Ref, Message, State}
+                    end;
                 not_found ->
                     %% something strange is happening, try again
                     take(ForActorID, State#state{pending_acks=maps:remove(ForActorID, State#state.pending_acks)})
```

Comment:
I'm just curious here but do you know what might trigger this strange behavior?

Reply:
I've never seen it happen, but it could be possible